cmd/tkey-sign/main.go

```diff
@@ -0,0 +1,389 @@
+	sign := pflag.BoolP("sign", "S", false, "Sign the message file")
+	verify := pflag.BoolP("verify", "V", false, "Verify the signature on the message file")
+	force := pflag.BoolP("force", "f", false, "Force writing of signature and pubkey files")
+	keyFile := pflag.StringP("public", "p", "", "Path to public key file.")
+	sigFile := pflag.StringP("sig", "x", "", "Path to signature file.")
+	messageFile := pflag.StringP("message", "m", "", "Message file")
+	devPath := pflag.StringP("port", "d", "",
+		"Set serial port device `PATH`. If this is not passed, auto-detection will be attempted.")
+	speed := pflag.IntP("speed", "s", tkeyclient.SerialSpeed,
+		"Set serial port speed in `BPS` (bits per second).")
+	enterUss := pflag.Bool("uss", false,
+		"Enable typing of a phrase to be hashed as the User Supplied Secret. The USS is loaded onto the TKey along with the app itself. A different USS results in different public/private keys, meaning a different identity.")
+	ussFile := pflag.String("uss-file", "",
+		"Read `FILE` and hash its contents as the USS. Use '-' (dash) to read from stdin. The full contents are hashed unmodified (e.g. newlines are not stripped).")
+	versionOnly := pflag.BoolP("version", "v", false, "Output version information.")
+	helpOnly := pflag.BoolP("help", "h", false, "Output this help.")
+
+	if version == "" {
+		version = readBuildInfo()
+	}
+
+	pflag.BoolVar(&verbose, "verbose", false, "Enable verbose output.")
+	pflag.Usage = func() {
+		desc := fmt.Sprintf(`Usage:
+
+%[1]s -h/--help
+
+%[1]s -G/--getkey [-d/--port device] [-f/--force] [-s/--speed speed] [--uss] [--uss-file ussfile] [--verbose] -p/--public pubkey
+
+%[1]s -S/--sign [-d/--port device] [-f/--force] -m message [-s speed] [--uss] [--uss-file ussfile] [--verbose] [-x sigfile]
+
+%[1]s -V/--verify -m message -p/--public pubkey [-x sigfile]
+
+%[1]s --version
+
+%[1]s signs or verifies the signature of a message in a file.
+The message will be hashed with SHA512 and either signed with Ed25519
+using the TKey's private key or verified given the public key.
+
+Exit status code is 0 if everything went well
+Otherwise exit code is non-zero.
+
+Alternatively, -G/--getkey can be used to receive the public key of
+the signer app on the TKey. Specify where to store it with -p`,
+			os.Args[0])
+
+		le.Printf("%s\n\n%s", desc,
+			pflag.CommandLine.FlagUsagesWrapped(86))
+	}
+
+	pflag.Parse()
+
+	if pflag.NArg() > 0 {
+		le.Printf("Unexpected argument: %s\n\n", strings.Join(pflag.Args(), " "))
+		pflag.Usage()
+		os.Exit(2)
+	}
+
+	if *versionOnly {
+		le.Printf("tkey-sign %s", version)
+		os.Exit(0)
+	}
+
+	if *helpOnly {
+		pflag.Usage()
+		os.Exit(0)
+
+	}
+
+	if *getKey {
+		cmd = cmdGetKey
+		cmdArgs++
+	}
+
+	if *sign {
+		cmd = cmdSign
+		cmdArgs++
+	}
+
+	if *verify {
+		cmd = cmdVerify
+		cmdArgs++
+	}
+
+	if cmdArgs > 1 {
+		pflag.Usage()
+		os.Exit(1)
+	}
+
+	switch cmd {
+	case cmdGetKey:
+		if *keyFile == "" {
+			le.Printf("Provide public key file with -p FILE")
```

Comment:
Here we say "Provide public key file with -p FILE".
While in the examples we say: "-p/--public pubkey"
and in the usage description we say: "-p, --public string"

We should use the same on all three instead of FILE, pubkey and string.

Reply:
I harmonized man page, help text and the error messages.